src/noc/containers/noc-field/form-validators.service.ts

```diff
@@ -0,0 +1,47 @@
+import { Injectable } from '@angular/core';
+import { FormControl, Validators } from '@angular/forms';
+import { NocQuestion } from '../../models';
+import { NocValidators } from './utils/noc-validators';
+
+@Injectable()
+export class FormValidatorsService {
+  private static readonly REGEX_WHITESPACES: string = '^[^ ]+(?:\\s+[^ ]+)*$';
+
+  public addValidators(nocQuestion: NocQuestion, control: FormControl): FormControl {
+    if (nocQuestion.answer_field_type.type) {
+      const validators = [];
+      if (nocQuestion.answer_field_type.type === 'Text') {
```

Comment:
A switch statement would be much cleaner, rather than a series of `else if`s.

Reply:
Agreed, but I would keep the same if/else block to align with ccd toolkit for further maintainence.